provider/disk_utils.py

```diff
@@ -0,0 +1,167 @@
+    """
+    This function provides the disk ID reference.
+    It will vary based on the OS (ID or path).
+
+    :param os_type: The VM Operating System.
+    :param disk_id: The VM disk ID defined in the test cfg.
+    :param session: The guest session.
+    """
+    if os_type == "windows":
+        idx_info = get_disk_props_by_serial_number(session, serial, ["Index"])
+        if idx_info:
+            return idx_info["Index"]
+    else:
+        return get_linux_drive_path(session, serial)
+
+
+def init_disk_by_id(
+    params,
+    vm,
+    did,
+    image_name,
+    fstype=None,
+):
+    """
+    Initializes the disk in the guest based on the received device ID.
+    By initialize, it means obtain the reference of the disk (ID or path) and
+    later in Linux systems, create the filesystem + mounting or formatting the
+    disk in Windows ones.
+
+    :param params: Dictionary with the test parameters.
+    :param vm: The VM object.
+    :param did: The ID of the VM's disk.
+    :param image_name: The image name. e.g: stg0
+    :param fstype: The kind of filesystem (ntfs or xfs).
+    """
+    session = vm.wait_for_login()
+    os_type = params.get("os_type")
+    dst_dir = params.get("dst_dir")
+    image_params = params.object_params(image_name)
+    img_size = image_params.get("image_size")
+
+    disk = get_disk_by_serial(os_type, image_name, session)
+
+    if os_type == "linux":
+        if not dst_dir:
+            dst_dir = f"/mnt/{image_name}"
+        partition_name = disk.split("/")[2]
+        utils_disk.create_filesyetem_linux(session, partition_name, fstype="xfs")
+        if utils_disk.mount(disk, dst_dir, fstype, session=session):
+            return dst_dir
+    else:
+        if dst_dir:
+            dst_dir = f"{did}:\\{dst_dir}"
+        else:
+            dst_dir = f"{did}:\\{image_name}"
+        utils_disk.update_windows_disk_attributes(session, disk)
+        utils_disk.clean_partition_windows(session, disk)
+        return utils_disk.configure_empty_disk(session, disk, img_size, "windows")[0]
+
+
+def execute_io_test(
+    test,
+    params,
+    vm,
+    image,
+    fstype=None,
+    dst_dir=None,
+    io_command=None,
+    serial=None,
+    clean=True,
+    ignore_all_errors=False,
+    timeout=60,
+):
+    """
+    Execute the I/O write test on the disk image inside the guest.
+    Note: The target disk image should be the uninitialized disk.
+
+    :param test: QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param vm: The VM object.
+    :param image: The image name. e.g: stg0
+    :param dst_dir: The destination directory for I/O write testing.
+    :param io_command: The command for I/O test.
+    :param serial: The serial number of the disk
+    :param clean: If True, cleans the environment
+    :param ignore_all_errors: Checks the errors of the I/O command
+    :param timeout: The timeout for the I/O command.
+    """
+
+    if fstype is None:
+        fstype = "xfs"
```

Comment:
for windows it is ntfs, for linux it is xfs

Reply:
Updated